javalib/src/main/scala/java/util/concurrent/atomic/AtomicLong.scala

```diff
@@ -1,10 +1,24 @@
 package java.util.concurrent.atomic
 
+import scala.scalanative.native
+import scala.scalanative.native.Atomic
+
 class AtomicLong(private[this] var value: Long)
     extends Number
     with Serializable {
   def this() = this(0L)
 
+  private val lock: native.Ptr[native.CInt] = native.stackalloc[native.CInt]
+  !lock = 0
+
+  private def getLock: Boolean = {
+    Atomic.compareAndSwapInt(lock, 0, 1)
+  }
+
+  private def releaseLock: Unit = {
+    !lock = 0
+  }
+
   final def get(): Long = value
```

Comment:
I curious if this will work correctly. My impression is that the JVM gives specific guarantees regarding access to memory, however, I am not sure that Scala Native provides any of those. It would be interesting to have a test, for example based on the example here: http://en.cppreference.com/w/c/language/atomic

I am also curious if it would be possible to somehow base the `AtomicX` implementation on primitives provided by `stdatomic.h` or if the penalty of boxing/unboxing etc would to big.